projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -360,12 +362,81 @@ export class IgxTreeGridComponent extends IgxGridBaseDirective implements GridTy
             this.loadChildrenOnRowExpansion(args);
         });
 
+        this.onRowAdded.pipe(takeUntil(this.destroy$)).subscribe(args => {
+            if (this.rowSelection === 'multipleCascade') {
+                let rec = this._gridAPI.get_rec_by_id(this.primaryKey ? args.data[this.primaryKey] : args.data);
+                if (rec && rec.parent) {
+                    this.gridAPI.grid.selectionService.updateCascadeSelectionOnFilterAndCRUD(
+                        new Set([rec.parent]), rec.parent.rowID);
+                } else {
+                    // The record is still not available
+                    // Wait for the change detection to update records through pipes
+                    requestAnimationFrame(() => {
+                        rec = this._gridAPI.get_rec_by_id(this.primaryKey ?
+                            args.data[this.primaryKey] : args.data);
+                        if (rec && rec.parent) {
+                            this.gridAPI.grid.selectionService.updateCascadeSelectionOnFilterAndCRUD(
+                                new Set([rec.parent]), rec.parent.rowID);
+                        }
+                        this.notifyChanges();
+                    });
+                }
+            }
+        });
+
+        this.onRowDeleted.pipe(takeUntil(this.destroy$)).subscribe(args => {
+            if (this.rowSelection === 'multipleCascade') {
+                if (args.data) {
+                    const rec = this._gridAPI.get_rec_by_id(
+                        this.primaryKey ? args.data[this.primaryKey] : args.data);
+                    this.handleCascadeSelection(args, rec);
+                } else {
+                    // if a row has been added and before commiting the transaction deleted
+                    const leafRowsDirectParents = new Set<any>();
+                    this.records.forEach(record => {
+                        if (record && !record.children && record.parent) {
+                            leafRowsDirectParents.add(record.parent);
+                        }
+                    });
+                    // Wait for the change detection to update records through pipes
+                    requestAnimationFrame(() => {
+                        this.gridAPI.grid.selectionService.updateCascadeSelectionOnFilterAndCRUD(leafRowsDirectParents);
+                        this.notifyChanges();
+                    });
+                }
+            }
+        });
+
+        this.onFilteringDone.pipe(takeUntil(this.destroy$)).subscribe(() => {
+            if (this.rowSelection === 'multipleCascade') {
```

Comment:
Checks should be performed against the enum, rather than against the string representation. This way we can change the string without having to find and replace it everywhere!